cypress/support/commands.ts

```diff
@@ -24,14 +24,107 @@
 //
 // -- This will overwrite an existing command --
 // Cypress.Commands.overwrite('visit', (originalFn, url, options) => { ... })
-//
-// declare global {
-//   namespace Cypress {
-//     interface Chainable {
-//       login(email: string, password: string): Chainable<void>
-//       drag(subject: string, options?: Partial<TypeOptions>): Chainable<Element>
-//       dismiss(subject: string, options?: Partial<TypeOptions>): Chainable<Element>
-//       visit(originalFn: CommandOriginalFn, url: string, options: Partial<VisitOptions>): Chainable<Element>
-//     }
-//   }
-// }
\ No newline at end of file
+
+declare global {
+  namespace Cypress {
+    interface Chainable {
+      // custom commands
+      createFlow(options: Partial<ICreateFlowOptions>): Chainable<void>,
+      addBlock(menuChoices: string[]): Chainable<string>,
+      selectBlock(uuid: string): Chainable<string>,
+      undo(): Chainable<void>,
+      redo(): Chainable<void>,
+      save(): Chainable<void>,
+      // connectExitToBlock(exitUuid: string, targetBlockUuid: string): Chainable<void>,
+      // dragBlock(blockUuid: string): Chainable<void>,
+    }
+  }
+}
+
+export interface ICreateFlowOptions {
+  label: string,
+  interactionTimeout: number,
+  languages: string[],
+  modes: string[],
+}
+
+Cypress.Commands.add('createFlow', (options: Partial<ICreateFlowOptions>) => {
+  const optionsWithDefaults: ICreateFlowOptions = {
+    label: 'Test flow',
+    interactionTimeout: 1000,
+    languages: ['English'],
+    modes: ['IVR', 'SMS', 'USSD'],
+    ...options,
+  }
+
+  cy.visit('http://localhost:8081')
+
+  cy.get('#app').should('exist')
+  cy.get('[data-cy="create-flow--btn"]').click()
+
+  cy.get('[data-cy="create--btn"]').click()
+
+  cy.get('[data-cy="flow-label--editor"]')
+    .find('textarea')
+    .type(optionsWithDefaults.label)
+
+  cy.get('[data-cy="interaction-timeout--editor"]')
+    .find('input')
+    .type(String(optionsWithDefaults.interactionTimeout))
+
+  for (const language of optionsWithDefaults.languages) {
+    cy.get('[data-cy="languages--selector"]').click()
+    cy.contains('.multiselect__option', language).click();
+  }
+
+  cy.get('[data-cy="create--btn"]').click()
+
+  // Let the first undo-redo snapshot be taken
+  cy.wait(500)
+})
+
+Cypress.Commands.add('addBlock', (menuChoices: string[]) => {
+  for (const choice of menuChoices) {
+    cy.get('[data-cy="blocks--menu"]')
+    cy.contains('[data-cy="blocks--menu-item"]', choice).click()
+  }
+
+  return cy.get('[data-cy^="block--"]').last().then((block) => {
+    return cy.wrap(block.attr('data-cy')!.replace('block--', ''))
+  })
```

Comment:
This relies on appending new blocks to the DOM, unsure how fragile this may be.

Reply:
could you pls leave this as a comment in the code, so dev could improve it on the fly in the future?